dds/idl/topic_keys.cpp

```diff
@@ -190,17 +191,42 @@ TopicKeys::Iterator::operator++()
       if (!struct_root) {
         throw Error(root_, "Invalid Key Iterator");
       }
-      ACE_CDR::ULong field_count = struct_root->nfields();
+      const ACE_CDR::ULong field_count = struct_root->nfields();
+      // If a nested struct marked as a key has no keys, all the fields are
+      // implied to be keys (expect those marked with @key(FALSE)).
+      if (pos_ == 0) {
+        if (level_ > 0) {
+          implied_keys_ = true;
+          for (; pos_ < field_count && implied_keys_; ++pos_) {
```

Comment:
```suggestion
          for (int field_idx = 0; field_idx < field_count && implied_keys_; ++field_idx) {
```
This pass doesn't need to touch `pos_`